kolibri/core/content/serializers.py

```diff
@@ -31,10 +31,20 @@ def __init__(self, *args, **kwargs):
 class ChannelMetadataSerializer(serializers.ModelSerializer):
     root = serializers.PrimaryKeyRelatedField(read_only=True)
     lang_code = serializers.SerializerMethodField()
+    progress_fraction = serializers.SerializerMethodField()
```

Comment:
We cache this request *forever* in the frontend: https://github.com/learningequality/kolibri/blob/develop/kolibri/core/content/api.py#L127 - meaning that this will never get updated. Also, an upstream cache would cache this, regardless of the user.

This is why we moved `progress_faction` off the ContentNodeViewset data, and instead made the separate `ContentNodeProgressViewset`.

We should fetch this data in the frontend the same way that we do the other content node progress, and just do an addiitonal non-blocking call to get this (and then dynamically update progress to be non-null when it does get set).

```
ContentNodeProgressResource.fetchModel(channel.root, { force: true}).then....
```

Reply:
Guessing that we probably want to do this for multiple channels at once, so can do `fetchCollection({ id__in: channels.map(c => c.root)});`

Reply:
Example here of how it was done for the recommended page: https://github.com/learningequality/kolibri/blob/44bf1ed3591321f4b7ef7c4e5c6ceda8c01db7ae/kolibri/plugins/learn/assets/src/views/RecommendedSubpage.vue#L88

With some guards against sending too many ids at once (which is something else that caused issues previously).